README.md

```diff
@@ -71,109 +113,88 @@ This script writes 5 files:
-are re-downloaded time and again (as many times as 240 in 4 month). The first
-step is to collect all URLs we have already downloaded in a previous batch.
-For example, if we already have the (deduplicated) URLs of the previous year,
-we should do something like this first:
-```
-zcat 2018/cc_index_dedup/*.gz | cut -d' ' -f2 | sort | gzip > 2018_index_urls.gz
-```
-or, parallelly,
+are re-downloaded time and again (as many times as 240 in 4 month). 
+
+If you have already downloaded previous batches, or have any other collection
+of URLs, collect all the url lists into a single directory and pass it on as
+the -s parameter.
+
+If you plan to download further batches it is higly recommended to add the
+url list of the current batch to that directory. You can do that by using the
+--export-urls-file argument and pass along a filename for the url list.
+
 ```
-ls *.gz | parallel -j12 -k "zcat {} | cut -d' ' -f2 " | sort | gzip > ../2018_index_urls.gz
+deduplicate_index_urls.py -i 02_index_filtered/2019-04 -o 03_index_dedup/2019-04 
+-s urls_already_collected/ -euf hu_hu_2019-04.gz -k biggest
 ```
 
-Alternatively, the URLs can be extracted from an already existing corpus:
+If you have not exported the url list, you can extract them at any later point:
 ```
 extract_attributes.py -i 2018/cc_corpus/ -a url -o 2018_urls.tsv.gz -P 12 2> /dev/null
 ```
 
-Generally, it is better to collect the list from the index, because the corpus
-might already have been filtered (based on language, length, etc.), which means
-a lot of URLs that we have downloaded and discarded don't appear in it, so we'll
-just re-download them again.
-
-Once the list of URLs to skip is complete, the index can be deduplicated like
-```
-deduplicate_index_urls.py -i 2019/cc_index_filtered/ -o 2019/cc_index_dedup/ -s 2018_index_urls.gz -k biggest
-```
+Generally, it is better to collect the url list from the index, and not the
+corpus. Because the corpus might already have been filtered (based on language, 
+length, etc.), which means a lot of URLs that we have downloaded and discarded 
+would no longer be included in it. So if we based our url list on the corpus we 
+would re-download unnecessary documents again.
 
-Note: unfortunately, loading the old URLs takes a very long time... also, the
-script runs in a single process as Python's shared memory performance is
+Note: unfortunately, loading the previously collected URLs takes a very long time... 
+also, the script runs in a single process as Python's shared memory performance is
 abysmal.
 
 ### Download pages
 
-Pages in the (filtered, deduplicated) index can be downloaded by the command
-```
-download_pages.py -o 2019/cc_downloaded -e warc.gz -i '2019/cc_index_dedup/*.gz'
-```
+Common crawl offers to ways to download their corpus: via http and via Amazon's
+S3 buckets. The first version is getting less and less support and since 2023 it
+has been throttled down. So we switched over to use the S3 version.
```

Comment:
we switched over -> we use. The code history is not important here (also, we started out with s3 originally).

Reply:
Ok.